### PR TITLE
Opt out of vmware testing rather than opt in

### PR DESCRIPTION
**Pull request by dirkmueller**
## Why is this PR needed?

Without this PR, VMware testing will be always skipped. 

Fixes https://github.com/SUSE/avant-garde/issues/1137

## What does this PR do?

We're saving the result of the filter-pr evaluation on whether it touches vmware
specific CI code paths inside the pipeline env, so that later pipeline steps can
reference the result. 

Also the logic is changed to be more aligned with "better be safe than sorry",
so it explicitely skips when it is determined to *not* match. 

## Info for QA

This is CI automation specific.

# Merge restrictions

(Please do not edit this)

We are in *v4-maintenance phase*, so we will restrict what can be merged to prevent unexpected surprises:

    What can be merged (merge criteria):
        2 approvals:
            1 developer: code is fine
            1 QA: QA is fine
        there is a PR for updating documentation (or a statement that this is not needed)

### CODE REVIEW

**Comment by tdaines42** [comment]
You might need to rebase on top of the latest master. Also I think according to https://github.com/SUSE/skuba/blob/master/CONTRIBUTING.md#git-commit-messages you need to remove the `Fixes SUSE/avant-garde#1137` from your commit message.

**Comment by dirkmueller** [comment]
@tdaines42 done

File: ci/jenkins/pipelines/prs/skuba-test-vmware.Jenkinsfile [hunk 1]
```diff
@@ -64,6 +62,12 @@ pipeline {
...
 
+        stage('Check for changes') { steps {
+            script {
+                env.shouldRun = !sh(script: "${PR_MANAGER} filter-pr --filename ${FILTER_SUBDIRECTORY}", returnStdout: true, label: "Filtering PR") ==~ /does not contain changes/
+            }
+        }}
+
         stage('Git Clone') { steps {
             deleteDir()
             checkout([$class: 'GitSCM',
```
Comment (pablochacin) [inline ci/jenkins/pipelines/prs/skuba-test-vmware.Jenkinsfile]: Just wondering if it makes sense to checkout the repo if we later decide not to execute any test.

File: ci/jenkins/pipelines/prs/helpers/pr-manager [hunk 2]
```diff
@@ -18,9 +18,10 @@ setup_python_env() {
     source ${VENVDIR}/bin/activate
     # Not calling to `pip3` directly because its shebang can exceed 128 characters
     # and this operation will fail. Check: https://github.com/pypa/pip/issues/1773
-    python -m pip install -r "${SDIR}/pr_manager/requirements.txt"
+    python -m pip -q --disable-pip-version-check install -U -r "${SDIR}/pr_manager/requirements.txt"
```
Comment (Klaven) [inline ci/jenkins/pipelines/prs/helpers/pr-manager]: I'm not sure I understand why this change was needed.

File: ci/jenkins/pipelines/prs/helpers/pr_manager/pr_manager.py [hunk 3]
```diff
@@ -59,12 +59,7 @@ def filter_pr(args):
         repo = g.get_repo(GITHUB_REPO)
 
         pull = repo.get_pull(CHANGE_ID)
-        changed_files = pull.get_files()
-        files_list = []
-
-        # change paginated list to normal list
-        for f in changed_files:
-            files_list.append(f.filename)
+        files_list = [f.filename for f in pull.get_files()]
```
Comment (Klaven) [inline ci/jenkins/pipelines/prs/helpers/pr_manager/pr_manager.py]: I generally like this change, just wondering what it has to do with the objective of this PR?

File: ci/jenkins/pipelines/prs/helpers/pr-manager (hunk 2 above)
Comment (Itxaka) [reply ci/jenkins/pipelines/prs/helpers/pr-manager]: I guess `-q` and `--disable-pip-version-check` is to avoid getting unnecessary output into the log?

Comment (Klaven) [reply ci/jenkins/pipelines/prs/helpers/pr-manager]: right. but that is not what this PR is about nor what the commit message talks about. I like to keep our PR's focused on what they said they changed. it makes going back though history easier.

Comment (Itxaka) [reply ci/jenkins/pipelines/prs/helpers/pr-manager]: Thats a really good point indeed, unrelated changes in the same commit/PR make baby yoda cry :cry:

File: ci/jenkins/pipelines/prs/helpers/pr_manager/pr_manager.py (hunk 3 above)
Comment (pablochacin) [reply ci/jenkins/pipelines/prs/helpers/pr_manager/pr_manager.py]: @Klaven I merged the PR before seeing your comment. Sorry :pray:

File: ci/jenkins/pipelines/prs/helpers/pr-manager (hunk 2 above)
Comment (pablochacin) [reply ci/jenkins/pipelines/prs/helpers/pr-manager]: At least, they should be in separated commits,

File: ci/jenkins/pipelines/prs/helpers/pr_manager/pr_manager.py (hunk 3 above)
Comment (Klaven) [reply ci/jenkins/pipelines/prs/helpers/pr_manager/pr_manager.py]: @pablochacin it's ok. I did not block because I'm not as involved in this project as you yet. so until I am I just comment :)

**Comment by dirkmueller** [comment]
Sorry, I will be a better git citizen going forward. I agree with the points that these were unrelated cleanups
